File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/NewHealthCalculators.kt
package com.qaantum.calculatorhub.calculators

// NEW HEALTH CALCULATORS

// Blood Sugar Converter
data class BloodSugarResult(val mgdl: Double, val mmol: Double, val category: String, val categoryColor: String)
class BloodSugarConverter {
    private val conversionFactor = 18.0182
    
    fun fromMgdl(mgdl: Double): BloodSugarResult {
        val mmol = mgdl / conversionFactor
        return BloodSugarResult(mgdl, mmol, getCategory(mgdl), getCategoryColor(mgdl))
    }
    
    fun fromMmol(mmol: Double): BloodSugarResult {
        val mgdl = mmol * conversionFactor
        return BloodSugarResult(mgdl, mmol, getCategory(mgdl), getCategoryColor(mgdl))
    }
    
    private fun getCategory(mgdl: Double) = when {
        mgdl < 70 -> "Low (Hypoglycemia)"
        mgdl < 100 -> "Normal (Fasting)"
        mgdl < 126 -> "Prediabetes (Fasting)"
        else -> "Diabetes Range"
    }
    
    private fun getCategoryColor(mgdl: Double) = when {
        mgdl < 70 -> "orange"
        mgdl < 100 -> "green"
        mgdl < 126 -> "amber"
        else -> "red"
    }
}

// VO2 Max Calculator
data class VO2MaxResult(val vo2max: Double, val fitnessLevel: String)
class VO2MaxCalculator {
    fun cooperTest(distanceMeters: Double): VO2MaxResult {
        val vo2max = (distanceMeters - 504.9) / 44.73
        return VO2MaxResult(vo2max, getFitnessLevel(vo2max))
    }
    
    fun heartRateMethod(maxHR: Double, restingHR: Double): VO2MaxResult {
        val vo2max = 15.3 * (maxHR / restingHR)
        return VO2MaxResult(vo2max, getFitnessLevel(vo2max))
    }
    
    private fun getFitnessLevel(vo2max: Double) = when {
        vo2max >= 50 -> "Excellent"
        vo2max >= 40 -> "Good"
        vo2max >= 30 -> "Average"
        vo2max >= 20 -> "Below Average"
        else -> "Poor"
    }
}

// Medication Dosage Calculator
data class MedicationDosageResult(val singleDose: Double, val volumePerDose: Double?, val dailyDose: Double)
class MedicationDosageCalculator {
    fun calculate(weightKg: Double, dosePerKg: Double, concentration: Double? = null, dosesPerDay: Int = 1): MedicationDosageResult {
        val singleDose = weightKg * dosePerKg
        val volumePerDose = concentration?.let { singleDose / it }
        val dailyDose = singleDose * dosesPerDay
        return MedicationDosageResult(singleDose, volumePerDose, dailyDose)
    }
    
    fun convertLbToKg(weightLb: Double) = weightLb * 0.453592
}

File: Calculator/ios-native/CalculatorHub/Calculators/NewHealthCalculators.swift
import Foundation

// NEW HEALTH CALCULATORS

// Blood Sugar Converter
struct BloodSugarResult { let mgdl, mmol: Double; let category, categoryColor: String }
class BloodSugarConverter {
    private let factor = 18.0182
    func fromMgdl(_ mgdl: Double) -> BloodSugarResult { BloodSugarResult(mgdl: mgdl, mmol: mgdl / factor, category: getCategory(mgdl), categoryColor: getColor(mgdl)) }
    func fromMmol(_ mmol: Double) -> BloodSugarResult { let mgdl = mmol * factor; return BloodSugarResult(mgdl: mgdl, mmol: mmol, category: getCategory(mgdl), categoryColor: getColor(mgdl)) }
    private func getCategory(_ mgdl: Double) -> String { switch mgdl { case ..<70: return "Low (Hypoglycemia)"; case 70..<100: return "Normal (Fasting)"; case 100..<126: return "Prediabetes (Fasting)"; default: return "Diabetes Range" } }
    private func getColor(_ mgdl: Double) -> String { switch mgdl { case ..<70: return "orange"; case 70..<100: return "green"; case 100..<126: return "amber"; default: return "red" } }
}

// VO2 Max Calculator
struct VO2MaxResult { let vo2max: Double; let fitnessLevel: String }
class VO2MaxCalculator {
    func cooperTest(distanceMeters: Double) -> VO2MaxResult { VO2MaxResult(vo2max: (distanceMeters - 504.9) / 44.73, fitnessLevel: getLevel((distanceMeters - 504.9) / 44.73)) }
    func heartRateMethod(maxHR: Double, restingHR: Double) -> VO2MaxResult { let v = 15.3 * (maxHR / restingHR); return VO2MaxResult(vo2max: v, fitnessLevel: getLevel(v)) }
    private func getLevel(_ v: Double) -> String { switch v { case 50...: return "Excellent"; case 40..<50: return "Good"; case 30..<40: return "Average"; case 20..<30: return "Below Average"; default: return "Poor" } }
}

// Medication Dosage Calculator
struct MedicationDosageResult { let singleDose: Double; let volumePerDose: Double?; let dailyDose: Double }
class MedicationDosageCalculator {
    func calculate(weightKg: Double, dosePerKg: Double, concentration: Double? = nil, dosesPerDay: Int = 1) -> MedicationDosageResult {
        let singleDose = weightKg * dosePerKg
        let volumePerDose = concentration.map { singleDose / $0 }
        return MedicationDosageResult(singleDose: singleDose, volumePerDose: volumePerDose, dailyDose: singleDose * Double(dosesPerDay))
    }
    func convertLbToKg(_ lb: Double) -> Double { lb * 0.453592 }
}
